test_install/test_install_cuda.cu

```diff
@@ -0,0 +1,367 @@
+    }
+
+    // core/preconditioner/ilu.hpp
+    {
+        /*auto test =*/gko::preconditioner::Ilu<>::build().on(cudaExec);
+    }
+
+    // core/preconditioner/jacobi.hpp
+    {
+        using Bj = gko::preconditioner::Jacobi<>;
+        /*auto test =*/
+        Bj::build().with_max_block_size(1u).on(cudaExec);
+    }
+
+    // core/solver/bicgstab.hpp
+    {
+        using Solver = gko::solver::Bicgstab<>;
+        /*auto test =*/Solver::build()
+            .with_criteria(
+                gko::stop::Iteration::build().with_max_iters(1u).on(cudaExec))
+            .on(cudaExec);
+    }
+
+    // core/solver/cg.hpp
+    {
+        using Solver = gko::solver::Cg<>;
+        /*auto test =*/Solver::build()
+            .with_criteria(
+                gko::stop::Iteration::build().with_max_iters(1u).on(cudaExec))
+            .on(cudaExec);
+    }
+
+    // core/solver/cgs.hpp
+    {
+        using Solver = gko::solver::Cgs<>;
+        /*auto test =*/Solver::build()
+            .with_criteria(
+                gko::stop::Iteration::build().with_max_iters(1u).on(cudaExec))
+            .on(cudaExec);
+    }
+
+    // core/solver/fcg.hpp
+    {
+        using Solver = gko::solver::Fcg<>;
+        /*auto test =*/Solver::build()
+            .with_criteria(
+                gko::stop::Iteration::build().with_max_iters(1u).on(cudaExec))
+            .on(cudaExec);
+    }
+
+    // core/solver/gmres.hpp
+    {
+        using Solver = gko::solver::Gmres<>;
+        /*auto test =*/Solver::build()
+            .with_criteria(
+                gko::stop::Iteration::build().with_max_iters(1u).on(cudaExec))
+            .on(cudaExec);
+    }
+
+    // core/solver/ir.hpp
+    {
+        using Solver = gko::solver::Ir<>;
+        /*auto test =*/Solver::build()
+            .with_criteria(
+                gko::stop::Iteration::build().with_max_iters(1u).on(cudaExec))
+            .on(cudaExec);
+    }
+
+    // core/solver/lower_trs.hpp
+    {
+        using Solver = gko::solver::LowerTrs<>;
+        /*auto test =*/Solver::build().on(cudaExec);
+    }
+
+    // core/stop/
+    {
+        // iteration.hpp
+        auto iteration =
+            gko::stop::Iteration::build().with_max_iters(1u).on(cudaExec);
+
+        // time.hpp
+        auto time = gko::stop::Time::build()
+                        .with_time_limit(std::chrono::milliseconds(10))
+                        .on(cudaExec);
+        // residual_norm_reduction.hpp
+        /*auto res_red =*/gko::stop::ResidualNormReduction<>::build()
+            .with_reduction_factor(1e-10)
+            .on(cudaExec);
+
+        // stopping_status.hpp
+        /*auto stop_status =*/gko::stopping_status{};
+
+        // combined.hpp
+        auto combined =
+            gko::stop::Combined::build()
+                .with_criteria(std::move(time), std::move(iteration))
+                .on(cudaExec);
+    }
+
+    std::cout << "test_install: the Ginkgo installation was correctly detected "
```

Comment:
Might be a good idea to mention that this is a test for CUDA:
```suggestion
    std::cout << "test_install_cuda: the Ginkgo installation was correctly detected "
```

Reply:
Good point, and we also need to execute it in the test_install target